src/_tkagg.cpp

```diff
@@ -203,9 +203,62 @@ static PyObject *_tkinit(PyObject *self, PyObject *args)
     return Py_None;
 }
 
+static PyObject *mpl_tk_blit(PyObject *self, PyObject *args) {
+    PyObject *py_photo = NULL,
+             *py_tk = NULL,
+             *py_interpaddr = NULL,
+             *py_photo_name = NULL;
+    int height, width;
+    unsigned char *data_ptr;
+    int x1, x2, y1, y2;
+    Tcl_Interp *interp;
+    char const *photo_name;
+    Tk_PhotoHandle photo;
+    Tk_PhotoImageBlock block;
+    if (!PyArg_ParseTuple(args, "O(iin)(iiii):blit",
+                          &py_photo,
+                          &height, &width, &data_ptr,
+                          &x1, &x2, &y1, &y2)) {
+        goto exit;
+    }
+    if (!(py_tk = PyObject_GetAttrString(py_photo, "tk")) ||
+        !(py_interpaddr = PyObject_CallMethod(py_tk, "interpaddr", NULL)) ||
+        !(interp = (Tcl_Interp *)PyLong_AsVoidPtr(py_interpaddr))) {
+        goto exit;
+    }
+    if (!(py_photo_name = PyObject_Str(py_photo)) ||
+        !(photo_name = PyUnicode_AsUTF8(py_photo_name)) ||
+        !(photo = TK_FIND_PHOTO(interp, photo_name))) {
+        // TK_FIND_PHOTO (obviously) does not set an exception on failure.
+        PyErr_SetString(PyExc_ValueError, "Failed to extract Tk_PhotoHandle");
+        goto exit;
+    }
+    block.pixelPtr = data_ptr + 4 * ((height - y2) * width + x1);
+    block.width = x2 - x1;
+    block.height = y2 - y1;
+    block.pitch = 4 * width;
+    block.pixelSize = 4;
+    block.offset[0] = 0;
```

Comment:
Can you use this to avoid the `buf.take` copy?

Reply:
done